### ISSUE TITLE
Cannot install redis after protobuf (php:8.0-alpine) – "Could not determine info from the PHP binary"

### ISSUE THREAD

**Issue by caugner** [mlocati/docker-php-extension-installer#302]
## Summary

When installing the `redis` extension after installing the `protobuf` extension, the installation fails with the following error:

```
In PHP.php line 275:

  Could not determine info from the PHP binary


install [--no-convert] [--defaults] [--source] [--with-configure-options WITH-CONFIGURE-OPTIONS] [--save-logs SAVE-LOGS] [--dry-run] [--php PHP] [--ini INI] [--tmp-dir TMP-DIR] [--version-override [VERSION-OVERRIDE]] [--] [<path>]
```

## Dockerfile

```Dockerfile
FROM php:8.0-cli-alpine

RUN apk add --no-cache git && \
    curl -sSL -o /usr/local/bin/install-php-extensions https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions && \
    chmod uga+x /usr/local/bin/install-php-extensions && \
    sync && \
    install-php-extensions protobuf redis && \
    rm /usr/local/bin/install-php-extensions
```

## Build log

```
Sending build context to Docker daemon  420.4kB

Step 1/2 : FROM php:8.0-cli-alpine
 ---> 9484fd084dd9
Step 2/2 : RUN apk add --no-cache git &&     curl -sSL -o /usr/local/bin/install-php-extensions https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions &&     chmod uga+x /usr/local/bin/install-php-extensions &&     sync &&     install-php-extensions protobuf redis &&     rm /usr/local/bin/install-php-extensions
 ---> Running in a68adde10ba8
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/x86_64/APKINDEX.tar.gz
(1/3) Installing expat (2.2.10-r1)
(2/3) Installing pcre2 (10.36-r0)
(3/3) Installing git (2.30.1-r0)
Executing busybox-1.32.1-r2.trigger
OK: 24 MiB in 35 packages
install-php-extensions v.1.2.13
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/x86_64/APKINDEX.tar.gz
v3.13.1-136-gc30d8ead50 [https://dl-cdn.alpinelinux.org/alpine/v3.13/main]
v3.13.1-136-gc30d8ead50 [https://dl-cdn.alpinelinux.org/alpine/v3.13/community]
OK: 13882 distinct packages available
### INSTALLING REQUIRED PACKAGES ###
# Packages to be kept after installation: zstd-libs
# Packages to be used only for installation: m4 libbz2 perl autoconf dpkg pkgconf dpkg-dev libmagic file libgcc libstdc++ binutils libgomp libatomic libgphobos gmp isl22 mpfr4 mpc1 gcc musl-dev libc-dev g++ perl-error perl-git git-perl make re2c zstd-dev
(1/30) Installing m4 (1.4.18-r2)
(2/30) Installing libbz2 (1.0.8-r1)
(3/30) Installing perl (5.32.0-r0)
(4/30) Installing autoconf (2.69-r3)
(5/30) Installing libgcc (10.2.1_pre1-r3)
(6/30) Installing libstdc++ (10.2.1_pre1-r3)
(7/30) Installing binutils (2.35.1-r1)
(8/30) Installing dpkg (1.20.6-r0)
(9/30) Installing pkgconf (1.7.3-r0)
(10/30) Installing dpkg-dev (1.20.6-r0)
(11/30) Installing libmagic (5.39-r0)
(12/30) Installing file (5.39-r0)
(13/30) Installing libgomp (10.2.1_pre1-r3)
(14/30) Installing libatomic (10.2.1_pre1-r3)
(15/30) Installing libgphobos (10.2.1_pre1-r3)
(16/30) Installing gmp (6.2.1-r0)
(17/30) Installing isl22 (0.22-r0)
(18/30) Installing mpfr4 (4.1.0-r0)
(19/30) Installing mpc1 (1.2.0-r0)
(20/30) Installing gcc (10.2.1_pre1-r3)
(21/30) Installing musl-dev (1.2.2-r0)
(22/30) Installing libc-dev (0.7.2-r3)
(23/30) Installing g++ (10.2.1_pre1-r3)
(24/30) Installing perl-error (0.17029-r1)
(25/30) Installing perl-git (2.30.1-r0)
(26/30) Installing git-perl (2.30.1-r0)
(27/30) Installing make (4.3-r0)
(28/30) Installing re2c (1.3-r1)
(29/30) Installing zstd-libs (1.4.5-r3)
(30/30) Installing zstd-dev (1.4.5-r3)
Executing busybox-1.32.1-r2.trigger
OK: 257 MiB in 65 packages
### INSTALLING REMOTE MODULE protobuf ###
  (installing version 3.13.0)
[91m  - Installing protobuf (3.13.0): [0m[91mDownloading (connecting...)[0m[91mDownloading (0%)[0m[91m           [0m[91mDownloading (25%)[0m[91mDownloading (30%)[0m[91mDownloading (55%)[0m[91mDownloading (85%)[0m[91mDownloading (100%)[0m[91m
[0m+-----------------------------------+----------+
| Package name                      | protobuf |
| Package version (current release) | 3.13.0   |
| Package status                    | stable   |
+-----------------------------------+----------+
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/x86_64/APKINDEX.tar.gz
(1/1) Installing .docker-php-ext-enable-deps (20210217.102553)
OK: 257 MiB in 66 packages
(1/1) Purging .docker-php-ext-enable-deps (20210217.102553)
OK: 257 MiB in 65 packages
### INSTALLING REMOTE MODULE redis ###

Warning: Missing arginfo for Google\Protobuf\OneofDescriptor::getName() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\OneofDescriptor::getField() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\OneofDescriptor::getFieldCount() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\EnumValueDescriptor::getName() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\EnumValueDescriptor::getNumber() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\EnumDescriptor::getPublicDescriptor() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\EnumDescriptor::getValueCount() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\EnumDescriptor::getValue() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Descriptor::getClass() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Descriptor::getFullName() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Descriptor::getField() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Descriptor::getFieldCount() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Descriptor::getOneofDecl() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Descriptor::getOneofDeclCount() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Descriptor::getPublicDescriptor() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\FieldDescriptor::getName() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\FieldDescriptor::getNumber() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\FieldDescriptor::getLabel() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\FieldDescriptor::getType() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\FieldDescriptor::isMap() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\FieldDescriptor::getEnumType() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\FieldDescriptor::getMessageType() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Internal\DescriptorPool::getGeneratedPool() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Internal\DescriptorPool::getDescriptorByClassName() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Internal\DescriptorPool::getDescriptorByProtoName() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Internal\DescriptorPool::getEnumDescriptorByClassName() in Unknown on line 0

Warning: Missing arginfo for Google\Protobuf\Internal\DescriptorPool::internalAddGeneratedFile() in Unknown on line 0
[91m
[0m[91mIn PHP.php line 275:
                                                
  Could not determine info from the PHP binary  
                                                

[0m[91minstall [--no-convert] [--defaults] [--source] [--with-configure-options WITH-CONFIGURE-OPTIONS] [--save-logs SAVE-LOGS] [--dry-run] [--php PHP] [--ini INI] [--tmp-dir TMP-DIR] [--version-override [VERSION-OVERRIDE]] [--] [<path>]
[0m[91m
[0m
```

PS: Note that the "Missing arginfo" warnings from protobuf will be fixed in the next release, see https://github.com/protocolbuffers/protobuf/issues/7721 and https://github.com/protocolbuffers/protobuf/pull/8105.

**Comment by mlocati** [comment]
This was a pickle problem, fixed by https://github.com/FriendsOfPHP/pickle/pull/226
I've just published pickle version 0.7.1 which included that fix, so everything should work now (see the test results of https://github.com/mlocati/docker-php-extension-installer/pull/303).

**Comment by caugner** [comment]
@mlocati Wow, that was quick. Indeed it works now, thanks a lot! 🚀
